Here is the very odd Linux error..
```
/usr/include/x86_64-linux-gnu/sys/types.h:146:10: error: 'stddef.h' file not found
#include <stddef.h>
         ^
<module-includes>:1:10: note: in file included from <module-includes>:1:
#include "AssertionReporting.h"
```

Looks like a bug?
http://askubuntu.com/questions/390805/clang-error-stddef-h-file-not-found
https://bugs.launchpad.net/ubuntu/+source/llvm-defaults/+bug/1242300
https://bugs.launchpad.net/ubuntu/+source/llvm-defaults/+bug/1198123
http://stackoverflow.com/questions/20587228/clang-error-stddef-file-not-found

--- a/include/swift/Runtime/Config.h
+++ b/include/swift/Runtime/Config.h
@@ -35,7 +35,7 @@
 /// Does the current Swift platform use LLVM's intrinsic "swiftcall"
 /// calling convention for Swift functions?
 #ifndef SWIFT_USE_SWIFTCALL
-#if __has_attribute(swiftcall)
+#if __has_attribute(swiftcall) || defined(__linux__)
 #define SWIFT_USE_SWIFTCALL 1
 #else
 #define SWIFT_USE_SWIFTCALL 0

--- a/stdlib/public/SwiftShims/SwiftStddef.h
+++ b/stdlib/public/SwiftShims/SwiftStddef.h
@@ -17,7 +17,11 @@
 // result, using stddef.h here would pull in Darwin module (which includes
 // libc). This creates a dependency cycle, so we can't use stddef.h in
 // SwiftShims.
-#if !defined(__APPLE__)
+// On Linux, the story is different. We get the error message
+// "/usr/include/x86_64-linux-gnu/sys/types.h:146:10: error: 'stddef.h' file not
+// found"
+// This is a known Clang/Ubuntu bug.
+#if !defined(__APPLE__) && !defined(__linux__)
 #include <stddef.h>
 typedef size_t __swift_size_t;
 #else

--- a/stdlib/public/SwiftShims/SwiftStddef.h
+++ b/stdlib/public/SwiftShims/SwiftStddef.h
@@ -13,7 +13,15 @@
 #ifndef SWIFT_STDLIB_SHIMS_SWIFT_STDDEF_H
 #define SWIFT_STDLIB_SHIMS_SWIFT_STDDEF_H
 
+// stddef.h is provided by Clang, but it dispatches to libc's stddef.h.  As a
+// result, using stddef.h here would pull in Darwin module (which includes
+// libc). This creates a dependency cycle, so we can't use stddef.h in
+// SwiftShims.
+#if !defined(__APPLE__)
+#include <stddef.h>
+typedef size_t __swift_size_t;
+#else
 typedef __SIZE_TYPE__ __swift_size_t;
+#endif
 
 #endif // SWIFT_STDLIB_SHIMS_SWIFT_STDDEF_H
-

--- a/stdlib/public/SwiftShims/SwiftStdint.h
+++ b/stdlib/public/SwiftShims/SwiftStdint.h
@@ -20,7 +20,19 @@
 
 // Clang has been defining __INTxx_TYPE__ macros for a long time.
 // __UINTxx_TYPE__ are defined only since Clang 3.5.
-
+#if !defined(__APPLE__)
+#include <stdint.h>
+typedef int64_t __swift_int64_t;
+typedef uint64_t __swift_uint64_t;
+typedef int32_t __swift_int32_t;
+typedef uint32_t __swift_uint32_t;
+typedef int16_t __swift_int16_t;
+typedef uint16_t __swift_uint16_t;
+typedef int8_t __swift_int8_t;
+typedef uint8_t __swift_uint8_t;
+typedef intptr_t __swift_intptr_t;
+typedef uintptr_t __swift_uintptr_t;
+#else
 typedef __INT64_TYPE__ __swift_int64_t;
 #ifdef __UINT64_TYPE__
 typedef __UINT64_TYPE__ __swift_uint64_t;
@@ -56,6 +68,6 @@ typedef unsigned __INT8_TYPE__ __swift_uint8_t;
 
 typedef __swift_intn_t(__INTPTR_WIDTH__) __swift_intptr_t;
 typedef __swift_uintn_t(__INTPTR_WIDTH__) __swift_uintptr_t;
+#endif
 
 #endif // SWIFT_STDLIB_SHIMS_SWIFT_STDINT_H
-

--- a/stdlib/public/SwiftShims/SwiftStdint.h
+++ b/stdlib/public/SwiftShims/SwiftStdint.h
@@ -15,12 +15,16 @@
 
 // stdint.h is provided by Clang, but it dispatches to libc's stdint.h.  As a
 // result, using stdint.h here would pull in Darwin module (which includes
-// libc).  This creates a dependency cycle, so we can't use stdint.h in
+// libc). This creates a dependency cycle, so we can't use stdint.h in
 // SwiftShims.
+// On Linux, the story is different. We get the error message
+// "/usr/include/x86_64-linux-gnu/sys/types.h:146:10: error: 'stddef.h' file not
+// found"
+// This is a known Clang/Ubuntu bug.
 
 // Clang has been defining __INTxx_TYPE__ macros for a long time.
 // __UINTxx_TYPE__ are defined only since Clang 3.5.
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) && !defined(__linux__)
 #include <stdint.h>
 typedef int64_t __swift_int64_t;
 typedef uint64_t __swift_uint64_t;

